torch/_refs/__init__.py

```diff
@@ -2392,6 +2399,67 @@ def permute(a: TensorLikeType, dims: DimsSequenceType) -> TensorLikeType:
     return prims.transpose(a, _permutation)
 
 
+@register_decomposition(torch.ops.aten.repeat)
+def repeat(a: Tensor, *repeat_shape) -> Tensor:
+    # NOTE: cannot use utils.extract_shape_from_varargs here
+    # because that also validates the shape, but the shape
+    # given to repeat may be "invalid"
+    if len(repeat_shape) == 1 and isinstance(repeat_shape[0], Sequence):
+        repeat_shape = tuple(repeat_shape[0])
```

Comment:
I know it's taken from the expand function. Let's modify `torch._prims_common.extract_shape_from_varargs` to have optional argument `validate_shape=True` and use it here.